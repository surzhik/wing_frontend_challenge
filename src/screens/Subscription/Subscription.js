import React from "react";
import SubscriptionSwitcher from "subscriptions/SubscriptionSwitcher";
import Box from "common/Box";
import Link from "common/Link";
import * as routes from "app/routes";
import add_green_circle from "common/img/add_green_circle.svg";
import styles from "./Subscription.module.css";
import * as PropTypes from "prop-types";

export class SubscriptionScreen extends React.Component {
  componentDidMount() {
    this.fetchData();
  }

  componentDidUpdate(oldProps) {
    if (this.props.subId !== oldProps.subId) {
      this.fetchData();
    }
  }

  fetchData = () => {
    let { subId, fetchFilteredContracts, subscription } = this.props;
    fetchFilteredContracts({ [subscription]: subId });
  };

  render() {
    const {
      subId,
      contracts,
      subActive,
      sku,
      navigateInsurancePlan,
      navigateInsurance,
      subscription
    } = this.props;

    return (
      <div className="Subscription">
        <SubscriptionSwitcher
          {...(subscription === "att_subscription"
            ? { attSubId: subId }
            : { sprintSubId: subId })}
          attRoute={routes.attSubscription}
          sprintRoute={routes.sprintSubscription}
        />
        <div className={styles.linkList}>
          {!contracts ||
            (contracts.length === 0 && (
              <Link
                className={styles.subscriptionLink}
                to={navigateInsurance(subId)}
              >
                <Box>
                  <img src={add_green_circle} />
                  Insurance
                </Box>
              </Link>
            ))}
          {subActive && (
            <Link
              className={styles.subscriptionLink}
              to={navigateInsurancePlan(subId, sku)}
            >
              <Box>
                <img src={add_green_circle} />
                Select plan
              </Box>
            </Link>
          )}
        </div>
      </div>
    );
  }
}

SubscriptionScreen.propTypes = {
  subId: PropTypes.any,
  contracts: PropTypes.array,
  subActive: PropTypes.bool,
  sku: PropTypes.string,
  fetchFilteredContracts: PropTypes.func,
  subscription: PropTypes.string,
  navigateInsurancePlan: PropTypes.func,
  navigateInsurance: PropTypes.func
};

export default SubscriptionScreen;
