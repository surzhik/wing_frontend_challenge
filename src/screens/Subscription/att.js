import { connect } from "react-redux";
import * as routes from "app/routes";

import Subscription from "./Subscription";
import { fetchFiltered as fetchFilteredContracts } from "insurance/insuranceContracts/actions";
import { getAttSubscription, getFilteredInsuranceContracts } from "reducers";

const mapStateToProps = (state, ownProps) => {
  const attSub = getAttSubscription(state, ownProps.subId);

  const filteredContracts = getFilteredInsuranceContracts(state, {
    att_subscription: ownProps.subId
  });

  return {
    subActive: attSub && attSub.att_status === "active",
    contracts: filteredContracts,
    sku: attSub && attSub.device_specs,
    navigateInsurancePlan: routes.attInsurancePlan,
    navigateInsurance: routes.attInsurance,
    subscription: 'att_subscription'
  };
};

const mapDispatchToProps = {
  fetchFilteredContracts
};

export default connect(mapStateToProps, mapDispatchToProps)(Subscription);
