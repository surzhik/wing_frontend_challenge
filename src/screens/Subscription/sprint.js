import { connect } from "react-redux";
import * as routes from "app/routes";
import { fetchFiltered as fetchFilteredContracts } from "insurance/insuranceContracts/actions";
import { getSprintSubscription, getFilteredInsuranceContracts } from "reducers";
import Subscription from "./Subscription";

const mapStateToProps = (state, ownProps) => {
  const sprintSub = getSprintSubscription(state, ownProps.subId);
  const filteredContracts = getFilteredInsuranceContracts(state, {
    subscription: ownProps.subId
  });

  return {
    subActive: sprintSub && sprintSub.sprint_status === "active",
    contracts: filteredContracts,
    sku: sprintSub && sprintSub.device_specs,
    navigateInsurancePlan: routes.sprintInsurancePlan,
    navigateInsurance: routes.sprintInsurance,
    subscription: "subscription"
  };
};

const mapDispatchToProps = {
  fetchFilteredContracts
};

export default connect(mapStateToProps, mapDispatchToProps)(Subscription);
