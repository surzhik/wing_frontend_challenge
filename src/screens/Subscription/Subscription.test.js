import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { shallow } from "enzyme";
import Subscription from "./Subscription";
import SubscriptionSwitcher from "subscriptions/SubscriptionSwitcher";
import Box from "common/Box";
const mockStore = configureMockStore();

describe("Subscription", () => {
  const props = {
    subActive: true,
    contracts: [{ id: 50 }],
    sku: "SOME",
    navigateInsurancePlan: jest.fn(),
    navigateInsurance: jest.fn(),
    fetchFilteredContracts: jest.fn()
  };

  describe("Subscription structure", () => {
    it("have unchanged structure", () => {
      const store = mockStore({});
      const wrapper = shallow(
        <Provider store={store}>
          <Subscription {...props} />
        </Provider>
      ).dive();
      expect(wrapper).toMatchSnapshot();
    });
  });

  const subscription = shallow(<Subscription {...props} />);
  it("should render", () => {
    expect(subscription).toHaveLength(1);
  });

  it("should have selector", () => {
    expect(subscription.find(SubscriptionSwitcher)).toHaveLength(1);
  });
  it("should have 1 boxes with current props", () => {
    expect(subscription.find(Box)).toHaveLength(1);
  });
});
