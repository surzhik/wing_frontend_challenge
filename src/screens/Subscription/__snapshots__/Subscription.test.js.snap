// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Subscription structure have unchanged structure 1`] = `
<div
  className="Subscription"
>
  <Connect(SubscriptionDecorator)
    attRoute={[Function]}
    sprintRoute={[Function]}
  />
  <div
    className="linkList"
  >
    <Connect(Link)
      className="subscriptionLink"
    >
      <Box>
        <img
          src="add_green_circle.svg"
        />
        Select plan
      </Box>
    </Connect(Link)>
  </div>
</div>
`;
